### PR TITLE
Set function

### PR DESCRIPTION
**Pull request by josephShokry**
adding the set function to the DB program. and the table & indices models.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+    // Use IntelliSense to learn about possible attributes.
```
Comment (mohamedabdelsalam) [inline .vscode/launch.json:2]: Does json accept comments !!!

File: source/commands_functions/command_factory.py [hunk 2]
```diff
@@ -13,6 +16,6 @@ def create_commands(args):
         elif command_type == "get":
             return GetCommand()
         elif command_type == "set":
-            return SetCommand()
+            return SetCommand(args.data_base, args.table, args.value, args.disableoverwrite, args.value_path)
```
Comment (mohamedabdelsalam) [inline source/commands_functions/command_factory.py:19]: Why you didn’t pass all the args ?

File: source/commands_functions/create_command.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import os, json
+from textwrap import indent
+from commands_functions.abstract_command import AbtractCommand
+from commands_functions.schema_keys import Keys
+from outputs.exceptions import *
+from outputs.status import Status
+
+
+
+def writeToJSONFile(path, fileName, data):
```
Comment (mohamedabdelsalam) [inline source/commands_functions/create_command.py:10]: Why this method is public. Also it doesn’t follow the convension

File: .vscode/launch.json (hunk 1 above)
Comment (mohamedabdelsalam) [inline .vscode/launch.json:2]: Don’t send a pr for for visual studio file

File: .vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+    "python.analysis.extraPaths": [
```
Comment (mohamedabdelsalam) [inline .vscode/settings.json:2]: Same above

File: source/commands_functions/create_command.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        filePathNameWExt =  path + '/' + fileName + '.json'
```
Comment (mohamedabdelsalam) [inline source/commands_functions/create_command.py:11]: Python convension isn’t a camel case. It is underscore

File: source/commands_functions/create_command.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        if not os.path.exists(schema_path) :
+            raise FileNotFound(status = Status.FileNotFound, message = "the schema path is not valid")
+        with open(schema_path, 'r') as schema: 
+            try:
+                schema_data = json.load(schema)
+            except:
+                raise WrongInput(status = Status.WrongInput, message = "schema is written in a wrong json format")
+        return schema_data
+
+    def execute(self):
```
Comment (mohamedabdelsalam) [inline source/commands_functions/create_command.py:33]: Create a class called 
```
    Database which has a Table object
Class Table should have TableMetadata
```

Comment (mohamedabdelsalam) [reply source/commands_functions/create_command.py:33]: All the following logic should be removed and move to Database class and Metadata class

Comment (mohamedabdelsalam) [reply source/commands_functions/create_command.py:33]: You can look at Mariam or Yahya repository to understand more. I already explained this in the session

File: source/models/indices.py [hunk 7]
```diff
@@ -0,0 +1,49 @@
+from abc import ABC, abstractmethod
```
Comment (mohamedabdelsalam) [inline source/models/indices.py:1]: File name
Should be index.py

File: source/models/indices.py [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+import os
+from outputs.exceptions import FileNotFound
+from models.table import Table
+
+class index(ABC):
```
Comment (mohamedabdelsalam) [inline source/models/indices.py:6]: This class should have serialize method

File: source/models/table.py [hunk 9]
```diff
@@ -0,0 +1,54 @@
+from abc import ABC
+import json, os
+from models.table_metadata import TableMetaData
+from outputs.exceptions import *
+
+
+class Table(TableMetaData):
```
Comment (mohamedabdelsalam) [inline source/models/table.py:7]: This class should have serialize method

File: source/models/table_metadata.py [hunk 10]
```diff
@@ -0,0 +1,36 @@
+from abc import ABC
+import json, os
+from commands_functions.schema_keys import Keys
+from outputs.exceptions import *
+
+class TableMetaData(ABC):
```
Comment (mohamedabdelsalam) [inline source/models/table_metadata.py:6]: This class should have serialize method

in createcommand you looop across the tables and say table.serialize($

File: source/commands_functions/command_factory.py (hunk 2 above)
Comment (josephShokry) [reply source/commands_functions/command_factory.py:19]: I preferred to send the args that every function need only.

File: source/models/indices.py (hunk 8 above)
Comment (josephShokry) [reply source/models/indices.py:6]: I think ```addIndices``` method is the serialize method just another name

File: source/models/table.py (hunk 9 above)
Comment (josephShokry) [reply source/models/table.py:7]: the same as above ```writeNewRow``` is the serialize method.
should I rename these functions?

**Pull request closed without merging**